### DISCUSSION TITLE
How to create a code exercise?

### DISCUSSION THREAD

**Discussion by pchalasani** [codewars/codewars.com#3107] [Q&A]
I want to create a series of koan/kata style staged exercises for people to learn how to use my open source code repo. 

I couldn't find a way to do this.

Any pointers appreciated.

**Comment by Oahaijil** [comment]
Recommended Tools: Use professional platforms (such as Codecademy's course tools, GitHub Classroom, or Exercism) to create staged exercises.

Steps for Creation:

Break down the source code from open-source repositories into smaller tasks of increasing difficulty.

Write clear objectives and test cases for each stage (pytest can be used).

Publish the tasks on the platform and guide learners to pull the code via Git, complete the tasks, and then submit.

**Reply by maderiaya** [reply]
This is a fantastic idea! "Koans" or "katas" are excellent for teaching the "flow" of using a codebase, as they force the learner to interact with the API in a structured way.
Since you couldn't find a specific tool that fits, it's likely because this is often achieved by combining existing patterns. Here are several approaches you can use to implement this in your open-source repository, ranging from simple to complex.
